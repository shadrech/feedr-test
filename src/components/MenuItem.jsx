import React from 'react';

export const MenuItem = ({ menuItem, dragRef, draggableProps }) => {
  return (
    <li ref={dragRef} {...draggableProps} className="item">
      <h2>{menuItem.name}</h2>
      <p>
        {menuItem.dietaries.map(diet => <span key={diet} className="dietary">{diet}</span>)}
      </p>
    </li>
  )
}
