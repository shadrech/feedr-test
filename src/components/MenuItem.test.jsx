import React from 'react'
import { shallow, configure } from 'enzyme';
import { MenuItem } from './MenuItem';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<MenuItem />', () => {

  it('renders a menu item correctly', () => {
    const item = {
      id: 1,
      name: 'Test item',
      dietaries: ['gf', 'df'],
    }
    const component = shallow(<MenuItem menuItem={item} draggableProps={{}} />)
    expect(component.html()).toMatchSnapshot()
  })

})
