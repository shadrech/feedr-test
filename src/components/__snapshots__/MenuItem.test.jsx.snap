// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> renders a menu item correctly 1`] = `"<li class=\\"item unselected\\"><h2>Test item</h2><p><span class=\\"dietary\\">gf</span><span class=\\"dietary\\">df</span></p></li>"`;
